text/src/autogluon/text/automm/predictor.py

```diff
@@ -569,21 +569,83 @@ def _fit(
 
         if trainer.global_rank == 0:
             top_k_avg_ckpt_path = os.path.join(save_path, "model.ckpt")
+            # Obtain the best K models sorted from the best performance to the worst performance
+            best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
+                                          key=lambda ele: ele[1], reverse=(minmax_mode == 'max'))
+            best_k_models_path = [ele[0] for ele in best_k_models_sorted]
+
+            # TODO(?) Currently, all top-K checkpoints are loaded to the main memory.
+            #  If the checkpoints are really large (e.g., super-giant backbones), this operation can take a lot of
+            #  CPU memory and may potentially trigger memory issue. Consider to optimize that using the
+            #  offload strategy.
             all_state_dicts, ckpt_template = gather_top_k_ckpts(
                 ckpt_dir=save_path,
-                ckpt_paths=checkpoint_callback.best_k_models.keys(),
+                ckpt_paths=best_k_models_path,
             )
+            if config.optimization.top_k_average_method == 'union_soup':
```

Comment:
Consider registering `union_soup`, `greedy_soup`, and `best` in constants.py and used the imported names.
 `max` and `min` are already defined in constants.py. Better to use their imported names.

Reply:
Done.